src/pages/home/report/ReportTypingIndicator.js

```diff
@@ -26,66 +26,52 @@ const defaultProps = {
     userTypingStatuses: {},
 };
 
-class ReportTypingIndicator extends React.Component {
-    constructor(props) {
-        super(props);
+const ReportTypingIndicator = (props) => {
+    const [usersTyping, setUsersTyping] = useState(props.userTypingStatuses ? _.filter(_.keys(props.userTypingStatuses), (login) => props.userTypingStatuses[login]) : []);
 
-        const usersTyping = props.userTypingStatuses ? _.filter(_.keys(props.userTypingStatuses), (login) => props.userTypingStatuses[login]) : [];
-        this.state = {usersTyping};
-    }
-
-    componentDidUpdate(prevProps) {
-        // Make sure we only update the state if there's been a change in who's typing.
-        if (_.isEqual(prevProps.userTypingStatuses, this.props.userTypingStatuses)) {
-            return;
-        }
-
-        const usersTyping = _.filter(_.keys(this.props.userTypingStatuses), (login) => this.props.userTypingStatuses[login]);
+    useEffect(() => {
+        setUsersTyping(_.filter(_.keys(props.userTypingStatuses), (login) => props.userTypingStatuses[login]));
```

Comment:
We're using `_.filter(_.keys(props.userTypingStatuses), (login) => props.userTypingStatuses[login])` twice, it can be a separate function

Reply:
@eVoloshchak I made a common function for getting userTypingStatuses values inside the ReportTypingIndicator Component and then I was getting a below hooks warning/error

```warning  React Hook useEffect has a missing dependency: 'getUserTypingStatuses'. Either include it or remove the dependency array  react-hooks/exhaustive-deps```

then, I extracted the userTypingStatuses function outside of the ReportTypingIndicator Component and passed userTypingStatuses props values as a param